src/pages/index.js

```diff
@@ -46,147 +55,240 @@ export const previewModalCloseButton = previewModal.querySelector(
+export const avatarButton = document.querySelector(".profile__image-button");
+export const editAvatarForm = document.querySelector(
+  "#edit-profile-picture-form"
+);
 
-// New Classes
+// New Classes /////////////////////////////////////////////////
 
+// New PopupConfirmation
+const popupConfirmation = new PopupConfirmation({
+  popupSelector: "#confirm-delete-modal",
+});
+popupConfirmation.setEventListeners();
+
+// New Class Edit Profile Popup
 const editProfilePopup = new PopupWithForm(
   { popupSelector: "#profile-edit-modal" },
   handleProfileEditSubmit
+  // handleProfileFormSubmit
 );
 editProfilePopup.setEventListeners();
 
+// New class New Card Popup
 const newCardPopup = new PopupWithForm(
   { popupSelector: "#add-card-modal" },
   handleAddNewCardSubmit
 );
 newCardPopup.setEventListeners();
 
+// New Class Image Preview Popup
 const imagePreviewPopup = new PopupWithImage({
   popupSelector: "#preview-image-modal",
 });
 imagePreviewPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+// New Class User Info
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
 
+// New Class Section
 const cardList = new Section(
   {
-    items: initialCards,
     renderer: (cardData) => {
-      const card = generateCard(cardData);
-      cardList.addItem(card);
+      createCard(cardData);
     },
   },
   ".cards__list"
 );
 
-// Profile Edit Modal
+// New Class Confirm Delete Popup
+const confirmDeleteModal = new PopupWithForm({
+  popupSelector: "#confirm-delete-modal",
+});
+confirmDeleteModal.setEventListeners();
+
+// New class New Form Validator
+const editFormValidator = new FormValidator(config, profileForm);
+editFormValidator.enableValidation();
+
+const addFormValidator = new FormValidator(config, addNewCardForm);
+addFormValidator.enableValidation();
+
+const avatarFormValidator = new FormValidator(config, editAvatarForm);
+avatarFormValidator.enableValidation();
+
+// New Class New Card
+function generateCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handlePreviewModal,
+    handleDeleteClick,
+    handleLikeIcon
+  );
+  return card.generateCard();
+}
+
+// Edit Avatar Modal///////////////////////////////////////////
+function handleAvatarSubmit(value) {
+  console.log(value);
+  //        Shows a loading indicator while updating the avatar.
+  avatarPopup.renderLoading(true);
+  //        Sends the new avatar to the backend via an API call.
+  api
+    //       Updates the UI if the request is successful.
+    .updateProfileAvatar(value.avatar)
+    .then((value) => {
+      userInfo.setAvatar(value.avatar);
+      //       Closes the popup upon success.
+      avatarPopup.close();
+      editAvatarForm.reset();
+      avatarFormValidator.disableButton();
+      avatarPopup.close();
```

Comment:
`      avatarPopup.close();
`

It's not needed here because you have it in line 166